packages/amplify-category-auth/resources/auth-custom-resource/hostedUIProviderLambda.js

```diff
@@ -1,102 +1,29 @@
-              authorize_scopes: providerMeta.authorize_scopes,
-            };
-          } else {
-            requestParams = null;
-          }
-        }
-        return requestParams;
-      };
-      let createIdentityProvider = (providerName) => {
-        let requestParams = getRequestParams(providerName);
-        if (!requestParams) {
-          return Promise.resolve();
-        }
-        requestParams.ProviderType = requestParams.ProviderName;
-        return identity.createIdentityProvider(requestParams).promise();
-      };
-      let updateIdentityProvider = (providerName) => {
-        let requestParams = getRequestParams(providerName);
-        if (!requestParams) {
-          return Promise.resolve();
-        }
-        return identity.updateIdentityProvider(requestParams).promise();
-      };
-      let deleteIdentityProvider = (providerName) => {
-        let params = { ProviderName: providerName, UserPoolId: userPoolId };
-        return identity.deleteIdentityProvider(params).promise();
-      };
-      let providerPromises = [];
-      identity
-        .listIdentityProviders({ UserPoolId: userPoolId, MaxResults: 60 })
-        .promise()
-        .then((result) => {
-          console.log(result);
-          let providerList = result.Providers.map((provider) => provider.ProviderName);
-          let providerListInParameters = hostedUIProviderMeta.map((provider) => provider.ProviderName);
-          hostedUIProviderMeta.forEach((providerMetadata) => {
-            if (providerList.indexOf(providerMetadata.ProviderName) > -1) {
-              providerPromises.push(updateIdentityProvider(providerMetadata.ProviderName));
-            } else {
-              providerPromises.push(createIdentityProvider(providerMetadata.ProviderName));
-            }
-          });
-          providerList.forEach((provider) => {
-            if (providerListInParameters.indexOf(provider) < 0) {
-              providerPromises.push(deleteIdentityProvider(provider));
-            }
-          });
-          return Promise.all(providerPromises);
-        })
-        .then(() => {
-          response.send(event, context, response.SUCCESS, {});
-        })
-        .catch((err) => {
-          console.log(err.stack);
-          response.send(event, context, response.FAILED, { err });
-        });
-    }
+
+    let deleteIdentityProvider = (providerName) => {
+      let params = { ProviderName: providerName, UserPoolId: userPoolId };
+      return identity.deleteIdentityProvider(params).promise();
+    };
+
+    let providerPromises = [];
+
+    hostedUIProviderMeta.forEach(({ ProviderName }) => providerPromises.push(deleteIdentityProvider(ProviderName)));
+
+    Promise.all(providerPromises)
+      .then(() => {
+        response.send(event, context, response.SUCCESS, {});
+      })
+      .catch((err) => {
+        console.log(err.stack);
+        response.send(event, context, response.FAILED, { err });
+      });
```

Comment:
Couple of things.

Is it ok to ignore `event.RequestType` ? It means this code will be executed at many stack events with the same inputs.
If this is desired then should this code be resilient to resource not found errors? I.e. if it attempts to delete already deleted resources.

Other unrelated consideration. If this lambda is to stay after Sep 2023 consider running it on nodejs18.x and using AWS SDK V3.

Reply:
an example of resource not found handling (there are couple of them in other categories).
https://github.com/aws-amplify/amplify-cli/blob/458f6414cd5a480fa01dda036e131260a9e466cb/packages/amplify-category-interactions/src/provider-utils/awscloudformation/function-template-dir/index.js.ejs#L16-L25

Reply:
TIL that if this does fail, the stack update breaks, so I will handle this error case (and make fixes to the other lambdas that I changed)